project/tests/com/google/daggerquery/executor/models/QueryTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package com.google.daggerquery.executor.models;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import org.junit.Test;
+
+public class QueryTest {
+  @Test
+  public void testParsingQuery_WithNullQueryType_ThrowsNullPointerException() {
+    try {
+      Query query = new Query(/*type = */ null, "com.google.cats.Cat");
+      fail();
+    } catch (NullPointerException e) {
```

Comment:
[`assertThrows`](https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html#assertThrows-java.lang.Class-org.junit.jupiter.api.function.Executable-) would be more idiomatic :smiley:

Reply:
The reason I chose this approach with try/catch is because `assertThrows` was supported only in JUnit 5.

As described in [blog](https://flexport.engineering/connecting-bazel-and-junit5-by-transforming-arguments-46440c6ea068), a solution for supporting JUnit 5 is following: adding `junit5.bzl` file from [here](https://github.com/junit-team/junit5-samples/tree/main/junit5-jupiter-starter-bazel) and using `java_junit5_test` instead of `java_test`.

I thought that it can make migration of project from open-source a little harder and will increase the total number of lines with this file `junit5.bzl` (which isn't a project enhancement).

How do you think?

Reply:
Ah sorry, I didn't realise this is a new feature, which seems to have been added since JUnit 4.13 (https://github.com/junit-team/junit4/blob/HEAD/doc/ReleaseNotes4.13.md).

Would it be straightforward to bump the JUnit version from 4.11 to 4.13 here:

https://github.com/googleinterns/dagger-query/blob/6f9f1b834ccf8876bbc4198d9132a3e39c06e21a/project/workspace_defs.bzl#L91

If not, how about using the `@Test(expected = ...Exception.class)` annotation, like this [StackOverflow post](https://stackoverflow.com/a/156528)? :smiley:

Reply:
For some reason changing version didn't help, but using this annotation with `expected` keyword is awesome, thanks! I didn't know about it before, tests look really better now 😊